frontend/resourceadm/components/ResourcePartyLists/ResourcePartyLists.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect, useState, useRef } from 'react';
+import { Link, useParams } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import { Alert, Button, Checkbox, Heading, Link as DigdirLink } from '@digdir/design-system-react';
+import classes from './ResourcePartyLists.module.css';
+import { useGetPartyListsQuery } from 'resourceadm/hooks/queries/useGetPartyLists';
+import { StudioSpinner } from '@studio/components';
+import { useGetResourcePartyListsQuery } from 'resourceadm/hooks/queries/useGetResourcePartyLists';
+import { useAddResourcePartyListMutation } from 'resourceadm/hooks/mutations/useAddResourcePartyListMutation';
+import { useRemoveResourcePartyListMutation } from 'resourceadm/hooks/mutations/useRemoveResourcePartyListMutation';
+import { getResourcePageURL } from 'resourceadm/utils/urlUtils';
+import { NewPartyListModal } from '../NewPartyListModal/NewPartyListModal';
+import { Resource } from 'app-shared/types/ResourceAdm';
+
+interface ResourcePartyListsProps {
+  env: string;
+  resourceData: Resource;
+}
+
+export const ResourcePartyLists = ({
+  env,
+  resourceData,
+}: ResourcePartyListsProps): React.ReactNode => {
+  const { t } = useTranslation();
+
+  const { selectedContext } = useParams();
+  const repo = `${selectedContext}-resources`;
+  const createPartyListModalRef = useRef<HTMLDialogElement>(null);
+
+  const [selectedLists, setSelectedLists] = useState<string[]>([]);
+
+  const {
+    data: envListData,
+    isLoading: isLoadingEnvListData,
+    error: envListDataError,
+  } = useGetPartyListsQuery(selectedContext, env);
+  const {
+    data: connectedLists,
+    isLoading: isLoadingConnectedLists,
+    error: connectedListsError,
+  } = useGetResourcePartyListsQuery(selectedContext, resourceData.identifier, env);
+  const { mutate: addResourcePartyList } = useAddResourcePartyListMutation(
+    selectedContext,
+    resourceData.identifier,
+    env,
+  );
+  const { mutate: removeResourcePartyList } = useRemoveResourcePartyListMutation(
+    selectedContext,
+    resourceData.identifier,
+    env,
+  );
+
+  useEffect(() => {
+    if (connectedLists) {
+      setSelectedLists(connectedLists.map((x) => x.partyListIdentifier));
+    }
+  }, [connectedLists]);
+
+  const handleRemove = (listItemId: string) => {
+    setSelectedLists((old) => old.filter((y) => y !== listItemId));
+    removeResourcePartyList(listItemId);
+  };
+
+  const handleAdd = (listItemId: string) => {
+    addResourcePartyList(listItemId);
+    setSelectedLists((old) => [...old, listItemId]);
+  };
+
+  if (isLoadingEnvListData || isLoadingConnectedLists) {
+    return <StudioSpinner spinnerText={t('general.loading')} />;
+  }
+
+  if (envListDataError || connectedListsError) {
+    return <Alert severity='danger'>{t('resourceadm.listadmin_load_list_error')}</Alert>;
```

Comment:
Can we add a test to check that the Alert is visible when it should and hidden when it should?